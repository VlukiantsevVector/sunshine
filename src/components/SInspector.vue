<template>
  <transition name="slide-in">
    <div v-if="show" :class="classes()">
      <form
        :class="classes('content')"
        v-bind="$attrs"
        v-on="$listeners"
      >
        <div :class="classes('header')">
          <slot name="header">
          </slot>
        </div>

        <div :class="classes('main')">
          <slot />
        </div>

        <div :class="classes('footer')">
          <slot name="footer">
          </slot>
        </div>
      </form>
    </div>
  </transition>
</template>

<script lang="ts">
import Vue from 'vue';
import mixins from 'vue-typed-mixins';
import { ClassesMixin } from '../lib/utils';

export default mixins(ClassesMixin).extend({
  name: 's-inspector',
  inheritAttrs: false,

  model: {
    prop: 'show',
    event: 'toggle',
  },

  props: {
    show: {
      type: Boolean,
      required: true,
    },
  },
});
</script>
