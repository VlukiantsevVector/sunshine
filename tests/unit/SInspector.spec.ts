import { mount, TransitionStub } from '@vue/test-utils';
import SInspector from '@/components/SInspector.vue';

describe('SInspector', () => {
  let wrapper : any;
  beforeAll(() => {
    wrapper = mount(SInspector, {
      stubs: {
        transition: TransitionStub,
      },
      propsData: {
        value: true,
      },
      slots: {
        header: 'Header',
        default: 'Content',
        footer: 'Footer',
      },
    });
  });

  it('inspector is visible and show header, content and footer content', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('inspector is not showing', () => {
    wrapper.setProps({ value: false });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
