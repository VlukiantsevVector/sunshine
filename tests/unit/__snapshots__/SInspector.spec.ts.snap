// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SInspector inspector is not showing 1`] = `undefined`;

exports[`SInspector inspector is visible and show header, content and footer content 1`] = `
<div class="s-inspector" name="slide-in">
  <form class="s-inspector__content">
    <div class="s-inspector__header">Header</div>
    <div class="s-inspector__main">Content</div>
    <div class="s-inspector__footer">Footer</div>
  </form>
</div>
`;
